pkg/ee/resource-quota/default-quota-controller/controller.go

```diff
@@ -230,3 +227,31 @@ func withSettingsEventFilter() predicate.Predicate {
 		},
 	}
 }
+
+func projectQuotaReconcilerFactory(resourceQuota *kubermaticv1.ResourceQuota) reconciling.NamedResourceQuotaReconcilerFactory {
+	return func() (string, reconciling.ResourceQuotaReconciler) {
+		return resourceQuota.Name, func(existing *kubermaticv1.ResourceQuota) (*kubermaticv1.ResourceQuota, error) {
+			existing.Spec = resourceQuota.Spec
+
+			if resourceQuota.Labels != nil {
+				if existing.Labels == nil {
+					existing.Labels = map[string]string{}
+				}
+				for k, v := range resourceQuota.Labels {
+					existing.Labels[k] = v
+				}
```

Comment:
Labels is a slice: won't this override other items in existing if the order is different?

What about

```
existing.Labels = sets.NewString(existing.Labels...).Insert(resourceQuota.Labels...).List()
```